util/cgiwifi.c

```diff
@@ -206,109 +254,152 @@ static void ICACHE_FLASH_ATTR reassTimerCb(void *arg) {
 		os_timer_arm(&resetTimer, 15000, 0); //time out after 15 secs of trying to connect
 	}
 }
+#else
+// ESP32
+wifi_config_t wifi_config;
+static void ICACHE_FLASH_ATTR startSta() {
+	//esp_wifi_stop();
+	//esp_wifi_set_mode(WIFI_MODE_STA);
+	esp_wifi_set_auto_connect(1);
+	esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);
+	esp_wifi_set_mode(WIFI_MODE_APSTA);
+	ESP_ERROR_CHECK( esp_wifi_connect());
+	//esp_wifi_start();
+	connTryStatus = CONNTRY_SUCCESS;
+	//httpd_printf("wifi_init_sta finished.\n");
+	//httpd_printf("connect to ap SSID:%s\n", (char *)wifi_config.sta.ssid);
+}
 #endif
 
 //This cgi uses the routines above to connect to a specific access point with the
 //given ESSID using the given password.
 CgiStatus ICACHE_FLASH_ATTR cgiWiFiConnect(HttpdConnData *connData) {
-#ifndef ESP32
 	char essid[128];
 	char passwd[128];
+#ifndef ESP32
 	static os_timer_t reassTimer;
-
-	if (connData->conn==NULL) {
+#endif
+	if (connData->isConnectionClosed) {
 		//Connection aborted. Clean up.
 		return HTTPD_CGI_DONE;
 	}
 
-	httpdFindArg(connData->post->buff, "essid", essid, sizeof(essid));
-	httpdFindArg(connData->post->buff, "passwd", passwd, sizeof(passwd));
-
+	httpdFindArg(connData->post.buff, "essid", essid, sizeof(essid));
+	httpdFindArg(connData->post.buff, "passwd", passwd, sizeof(passwd));
+#ifndef ESP32
 	strncpy((char*)stconf.ssid, essid, 32);
 	strncpy((char*)stconf.password, passwd, 64);
-	httpd_printf("Try to connect to AP %s pw %s\n", essid, passwd);
 
 	//Schedule disconnect/connect
 	os_timer_disarm(&reassTimer);
 	os_timer_setfn(&reassTimer, reassTimerCb, NULL);
 //Set to 0 if you want to disable the actual reconnecting bit
+#else
+	strncpy((char*)wifi_config.sta.ssid, essid, 32);
+	strncpy((char*)wifi_config.sta.password, passwd, 64);
+#endif
+	//httpd_printf("Try to connect to AP %s pw %s\n", essid, passwd);
+	connTryStatus = CONNTRY_WORKING;
 #ifdef DEMO_MODE
 	httpdRedirect(connData, "/wifi");
 #else
+#ifndef ESP32
 	os_timer_arm(&reassTimer, 500, 0);
 	httpdRedirect(connData, "connecting.html");
+#else
+	startSta();
+	httpdRedirect(connData, "connecting.html");
 #endif
+
 #endif
+
 	return HTTPD_CGI_DONE;
 }
 
 //This cgi uses the routines above to connect to a specific access point with the
 //given ESSID using the given password.
 CgiStatus ICACHE_FLASH_ATTR cgiWiFiSetMode(HttpdConnData *connData) {
-#ifndef ESP32
 	int len;
 	char buff[1024];
 
-	if (connData->conn==NULL) {
+	if (connData->isConnectionClosed) {
 		//Connection aborted. Clean up.
 		return HTTPD_CGI_DONE;
 	}
 
 	len=httpdFindArg(connData->getArgs, "mode", buff, sizeof(buff));
 	if (len!=0) {
-		httpd_printf("cgiWifiSetMode: %s\n", buff);
+		//httpd_printf("cgiWifiSetMode: %s\n", buff);
```

Comment:
why are the httpd_printf() calls being removed? Can the ESP_LOGx() calls be used here instead?

Reply:
Looks like platform.h only defines httpd_printf() when FREERTOS is not defined. I've replaced with ESP_LOGx within ESP32 specific sections.